picard/similarity.py

```diff
@@ -53,12 +53,22 @@ def similarity(a1, b1):
 
 def similarity2(a, b):
     """Calculates similarity of a multi-word strings."""
+    if not a or not b:
+        return 0.0
+    if a == b:
+        return 1.0
+
     alist = list(filter(bool, _split_words_re.split(a.lower())))
     blist = list(filter(bool, _split_words_re.split(b.lower())))
-    total = 0
-    score = 0.0
-    if len(alist) > len(blist):
+
+    alen, blen = len(alist), len(blist)
+    if not alen or not blen:
+        return 0.0
```

Comment:
Here again, what is expected if both are 0?

Reply:
Check my first commit, it had tests before I made any change, I preserved the previous behavior, so if a or b (or both) are empty, it returns 0.0.